src/main/java/uk/gov/hmcts/darts/task/runner/impl/UnstructuredDataExpiryDeletionAutomatedTask.java

```diff
@@ -0,0 +1,165 @@
+        deleteMediaEntity(userAccount, maxRetentionDate, batchSize);
+        deleteAnnotationDocumentEntity(userAccount, maxRetentionDate, batchSize);
+        deleteCaseDocumentEntity(userAccount, maxRetentionDate, batchSize);
+    }
+
+    void deleteTranscriptionDocumentEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            transcriptionDocumentRepository,
+            externalObjectDirectoryRepository.findExpiredTranscriptionDocuments(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getTranscriptionDocumentEntity
+        );
+    }
+
+
+    void deleteMediaEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            mediaRepository,
+            externalObjectDirectoryRepository.findExpiredMediaEntries(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getMedia
+        );
+    }
+
+    void deleteAnnotationDocumentEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            annotationDocumentRepository,
+            externalObjectDirectoryRepository.findExpiredAnnotationDocuments(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getAnnotationDocumentEntity
+        );
+    }
+
+    void deleteCaseDocumentEntity(UserAccountEntity userAccount, OffsetDateTime maxRetentionDate, Limit batchSize) {
+        deleteExternalObjectDirectoryEntity(
+            userAccount,
+            caseDocumentRepository,
+            externalObjectDirectoryRepository.findExpiredCaseDocuments(maxRetentionDate, batchSize),
+            ExternalObjectDirectoryEntity::getCaseDocument
+        );
+    }
+
+
+    <T extends SoftDelete> void deleteExternalObjectDirectoryEntity(
+        UserAccountEntity userAccount,
+        JpaRepository<T, Integer> repository,
+        List<ExternalObjectDirectoryEntity> externalObjectDirectoryEntities,
+        Function<ExternalObjectDirectoryEntity, T> entityMapper) {
+
+
+        List<ExternalObjectDirectoryEntity> deletedEntities = externalObjectDirectoryEntities
+            .stream()
+            .filter(this::deleteFromExternalDataStore)
+            .toList();
+
+        repository.saveAll(
+            deletedEntities
+                .stream()
+                .map(entityMapper)
+                .peek(entity -> entity.markAsDeleted(userAccount))
+                .toList()
+        );
+        externalObjectDirectoryRepository.deleteAll(deletedEntities);
+    }
+
+    boolean deleteFromExternalDataStore(ExternalObjectDirectoryEntity externalObjectDirectoryEntity) {
+        if (ExternalLocationTypeEnum.INBOUND.getId().equals(externalObjectDirectoryEntity.getExternalLocationType().getId())) {
```

Comment:
Do we need outbound call here?

Reply:
Also before we call the delete on the inbound and unstructured the ticket mentions:-

For inbound and unstructured, make sure there is an entry for ARM (elt_id = 3) in the external_object_directory table with a stored status (osr_id = 2)

Reply:
This logic is done in the select statement  to locate data to be deleted